### ISSUE TITLE
Conversations not working in background on Galaxy J3

### ISSUE THREAD

**Issue by sledz** [siacs/Conversations#3003]
#### General information

* **Version:** 2.1.2
* **Device:** Samsung Galaxy J3 SM-J330F
* **Android Version:** Android 7.0 Samsung
* **Samsung Experience-Version:** 8.1
* **Knox:** 2.8
* **Server name:** jabber.in-berlin.de
* **Server software:** ejabberd ???
* **Installed server modules:** XEP-0163, XEP-0191, XEP-0198, XEP-0237, XEP-0280, XEP-0313, XEP-0352 (reported OK by Conversations Server Info)
* **Conversations source:** F-Droid

Conversations does not receive messages when the app is in background. In this case on the sender side the message is sent, stays in status "waiting" and may need up to several hours to arrive at the other end. When both apps are in foreground during communication the messages arrive immediately and with a delay of only a few seconds.

Same account/server setup is working fine with other devices like Fairphone FP2.

**Comment by licaon-kter** [comment]
You've enabled _Foreground service_?

**Comment by sledz** [comment]
_Battery optimizations_ is disabled.

_Foreground service_ is not yet enabled.

**Comment by licaon-kter** [comment]
Enable it and see how it goes for a while.

**Comment by sledz** [comment]
OK, I'll try.

**Comment by sledz** [comment]
Checked! With enabled _Foreground service_ everything works fine.

**Comment by tyler71** [comment]
Using J3 (Android 7.0) as well, I did not need to enable the foreground service.
There seems to be two locations to disable power saving for a app, and they seem very similar in purpose.
Previously I only did the first one and I needed to enable the foreground service.

* Settings -> Device Maintenance -> Battery -> Unmonitored apps -> Add Apps (Conversations)
* Settings -> Device Maintenance -> Battery -> Battery Usage -> Optimize Battery Usage -> Apps not optimized (Conversations)
